### DISCUSSION TITLE
How to change the MQTT discovery prefix?

### DISCUSSION THREAD

**Discussion by roblomq** [arendst/Tasmota#16063] [Q&A] [answered]
Is it possible to change the default "homeassistant" MQTT discovery prefix to something else? I can't find a setting somewhere.
I don't use homeassistant but do use Domoticz. As Domoticz also has a MQTT autodiscovery I would like to use this native autodiscovery  . As I have several devices it would be more convenient when I change the prefix to Domoticz instead of Homeassistant.

**Comment by sfromis** [comment, marked as answer]
Ideally, you should no longer use it, as it is a deprecated outdated feature, not getting maintenance. Anyway:
https://github.com/arendst/Tasmota/blob/development/tasmota/my_user_config.h#L434-L438

**Reply by barbudor** [reply]
Tasmota also has a native support for Domoticz as this is Theo's choice of automation

**Reply by roblomq** [reply]
I understand its a choice but might be reconsidered maybe? The autodiscovery in Domoticz is quite new and now native. At the current choice of automation the user needs to create virtual devices and link them. While using the MQTT autodiscovery everything goes automatic. This works then the same way for HomeAssistant and Domoticz and maybe others too. Any I assume the MQTT functionality is getting maintenance so it only needs a discovery prefix, or am I missing something?

**Reply by sfromis** [reply]
Well, I specifically told you that the outdated MQTT Discovery code in Tasmota is out of maintenance, and it is no longer included in the precompiled Tasmota builds. In principle, someone could submit a PR, but that does not mean that it is really included in regular maintenance. While Tasmota is certainly continuing to use MQTT, that is really unrelated to the "MQTT autodiscovery" feature of HA.

Using a deprecated feature which is strongly tied to the internal structure of Home Assistant was IMO not a great strategy. The old way was deprecated as not working well, and replaced by the newer Tasmota integration which shifts the dependence on internal Home Assistant structure out of Tasmota.

**Reply by roblomq** [reply]
It's clear, this issue can be closed then.
